src/test/Jayz.test.ts

```diff
@@ -24,7 +24,7 @@ describe("JayZ", () => {
     const encryptedItem = await jayz.encryptItem({
       item: bankAccount,
       fieldsToEncrypt
-    })
+    }) as any
```

Comment:
`encryptedItem` is of type `ItemWithoutEncryptedFields`, plus a metadata object. `ItemWithoutEncryptedFields` is a subset of the fields in `item`. 

However, I'm not able to call `encryptedItem`.`{key that exists on item}` without throwing an error: `Property 'pk' does not exist on type 'EncryptedJayZItem<BankAccount, keyof BankAccount>'.`

Because of that error, I casted this `as any`. I would prefer not to do that if anyone knows how to properly define the `ItemWithoutEncryptedFields` type.

Reply:
Hey @kedskeds, 

The reason you currently need that `as any` cast is because of the  `(keyof BankAccount)[] ` type annotation attached to the `fieldsToEncrypt` declaration on  line 15.

What that's doing is telling TypeScript that this array can contain _any_ key of `BankAccount`, and when you hit the `Omit<T, K>` bits in the encryption types that's interpreted to omit _every_ key from the type.

To Fix that, there's a few small things you'd need to do: 

Change this:

```TypeScript
const fieldsToEncrypt: (keyof BankAccount)[] = [
    "accountNumber",
    "balance",
    "routingNumber",
    "notes"
  ]
```

To this: 
```TypeScript
const fieldsToEncrypt = [
    "accountNumber",
    "balance",
    "routingNumber",
    "notes"
  ] as const
```

The `as const` will make TypeScript _narrow_ the type to the "literal" contents of the array, i.e. `readonly ["accountNumber", "balance", "routingNumber", "notes"]`

Note how that also appends a `readonly` modifier to the inferred type. So you'll need to also update `fieldsToEncrypt: K[]` declarations in the EncryptParams to `fieldsToEncrypt: readonly K[]` to stop TS complaining about readonly and mutable types being incompatible.

After that you'll get the "correct" type inferred. You'll still have a few type errors on lines 21+ since any field that's "encrypted" (e.g. `accountNumber`) won't show up as a property on the object (which is intended) and thus TS will complain -- so you'd just need to tweak that a bit.

Reply:
Thank you for this excellent explanation @joshc-hh!!